hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/DatanodeStateMachine.java

```diff
@@ -242,7 +241,7 @@ private void start() throws IOException {
           try {
             Thread.sleep(nextHB.get() - now);
           } catch (InterruptedException e) {
-            LOG.warn("Interrupt the execution.", e);
+            //triggerHeartbeat is called during the sleep
```

Comment:
Hmm I might misunderstand this: will then the InterruptedException silently be hidden? If so is this expected behavior?

Reply:
Good question. Based on my understanding the interrupt is intentional (when DN would like to send out a HB immediately) 

```java
 /**
   * Calling this will immediately trigger a heartbeat to the SCMs.
   * This heartbeat will also include all the reports which are ready to
   * be sent by datanode.
   */
  public void triggerHeartbeat() {
    if (stateMachineThread != null) {
      stateMachineThread.interrupt();
    }
  }
```

I think it's expected to interrupt the thread, therefore we need to hide the exception.

Reply:
Got it! Thanks for your clarification.